src/dpq2/oids.d

```diff
@@ -156,15 +156,18 @@ OidType detectOidTypeFromNative(T)()
 {
     import std.datetime.date : StdDate = Date, TimeOfDay;
     import std.datetime.systime : SysTime;
-    import std.traits : OriginalType, TemplateArgsOf, Unqual;
+    import std.traits : isIntegral, OriginalType, TemplateArgsOf, Unqual;
     import std.typecons : Nullable;
     static import dpq2.conv.time;
     import vibe.data.json : VibeJson = Json;
 
     static if(is(Unqual!T == Nullable!R,R))
         return detectOidTypeFromNative!(TemplateArgsOf!T[0]); //remove Nullable from Type before test
     else static if (is(Unqual!T == enum))
+    {
+        static assert(isIntegral!T || is(OriginalType!T == string), "Only integral and string enums are supported");
```

Comment:
This brings us back to the problem of `long` type support.

Can you add check of support long enums sized more than `int.max` into `~master`? Then we can simple merge it into one of theese PRs.

Reply:
see #97